src/runtime/components/server/runtime_support_server.cpp

```diff
@@ -1581,13 +1580,12 @@ namespace hpx { namespace components { namespace server
                 typedef boost::tokenizer<boost::char_separator<char> > tokenizer;
                 boost::char_separator<char> sep(HPX_INI_PATH_DELIMITER);
                 tokenizer tokens(component_path, sep);
-                boost::system::error_code fsec;
                 for(tokenizer::iterator it = tokens.begin(); it != tokens.end(); ++it)
                 {
                     lib = fs::path(*it);
                     fs::path lib_path =
                         lib / std::string(HPX_MAKE_DLL_STRING(component));
-                    if(fs::exists(lib_path, fsec))
+                    if(fs::exists(lib_path))
```

Comment:
This will throw an exception, no? I think for the purpose of the filesystem API we can introduce a `filesystem::error_code` that maps onto the appropriate type (`std` or `boost`) and use that for the filesystem API calls we have.